src/server/tiered_storage.h

```diff
@@ -105,6 +107,9 @@ class TieredStorage {
 
   PrimeTable::Cursor offloading_cursor_{};  // where RunOffloading left off
 
+  // Stash operations waiting for completion to throttle
+  tiering::EntryMap<::util::fb2::Future<bool>> stash_backpressure_;
```

Comment:
Either a future or... a blocking counter? I don't see how we can easily use something more lightweight